src/components/item/form/DocumentForm.tsx

```diff
@@ -100,27 +99,25 @@ export const DocumentExtraForm = ({
   );
 };
 
-type Props = {
-  onChange: BaseFormProps['onChange'];
-  item?: DocumentItemTypeRecord;
-  updatedProperties: Partial<DocumentItemType>;
-};
-
 const DocumentForm = ({
-  onChange,
+  setChanges,
   item,
   updatedProperties,
-}: Props): JSX.Element => {
+}: EditModalContentPropType): JSX.Element => {
   const { t: translateBuilder } = useBuilderTranslation();
 
+  // we cast the properties here to match the generic component EditModalContentType
+  const typedUpdatedProperties = updatedProperties as Partial<DocumentItemType>;
+  const typedItem = item as DocumentItemType;
```

Comment:
Is there really no other way than casting ?

Reply:
Even with generic I can't sorry ;(